src/test/scala/com/lucidworks/spark/EventsimTestSuite.scala

```diff
@@ -118,6 +118,40 @@ class EventsimTestSuite extends EventsimBuilder {
     assert(queryDF.count() == eventSimCount)
   }
 
+  test("Timestamp filter queries") {
+    val df: DataFrame = sqlContext.read.format("solr")
+      .option("zkHost", zkHost)
+      .option("collection", collectionName)
+      .load()
+    df.registerTempTable("events")
+
+    val timeQueryDF = sqlContext.sql("SELECT * from events WHERE `registration` = '2015-05-31 11:03:07Z'")
+    assert(timeQueryDF.count() == 11)
+  }
+
+  test("Length range filter queries") {
+    val df: DataFrame = sqlContext.read.format("solr")
+      .option("zkHost", zkHost)
+      .option("collection", collectionName)
+      .load()
+    df.registerTempTable("events")
+
+    val timeQueryDF = sqlContext.sql("SELECT * from events WHERE `length` >= '700' and `length` <= '1000'")
+    assert(timeQueryDF.count() == 1)
+  }
+
+  // Ignored since Spark is not passing timestamps filters to the buildScan method. Range timestamp filtering is being done at Spark layer
+  ignore("Timestamp range filter queries") {
```

Comment:
While working on this, we found something interesting. Spark does not push down the range timestamp filters to the dataSource level. The timestamp ranges are always performed as the Spark layer which is inefficient. 

The range filters for other types works fine